frontend/api/users/useAPISettingsPassword.ts

```diff
@@ -0,0 +1,62 @@
+import { BACKEND_URL } from "../../config";
+import { useMutation } from "@tanstack/react-query";
+import { useRouter } from "next/router";
+import { FetchErrorsType, FetchUrl } from "../types/Fetch";
+
+interface ResponseType {
+    errors?: string[];
+    data?: unknown;
+}
+
+export interface APISettingsMutationVariables {
+    userId: string;
+    old_password: string;
+    new_password: string;
+    repeat_password: string;
+}
+
+export const useAPISettingsPassword = () => {
+    const { push } = useRouter();
+
+    const oldValues = {
+        userId: "63b325e386f12b8784f47c93", //Na razie na sucho Id i pozostałe dane
+        password: "Ab1CdE",
+    };
+
+    return useMutation<
+        ResponseType,
+        FetchErrorsType,
+        APISettingsMutationVariables
+    >(async ({ old_password, new_password, repeat_password }) => {
+        if (new_password !== repeat_password) {
```

Comment:
checking for these things should be handled on backend side, if you want to also check it client side I'll send you that tomorrow